Tools/cases_generator/analyzer.py

```diff
@@ -321,11 +328,25 @@ def analyze_stack(
     ]
     # Mark variables with matching names at the base of the stack as "peek"
     modified = False
-    for input, output in zip(inputs, outputs):
-        if input.name == output.name and not modified:
-            input.peek = output.peek = True
+    input_names: dict[str, lexer.Token] = { i.name : i.first_token for i in op.inputs if i.name != "unused" }
+    input_names
+    for input, output in itertools.zip_longest(inputs, outputs):
+        if output is None:
+            pass
+        elif input is None:
+            if output.name in input_names:
+                raise analysis_error(
+                    f"Reuse of variable '{output.name}' at different stack location",
```

Comment:
```suggestion
                    f"Reuse of variable {output.name!r} at different stack location",
```

(Same suggestion below; this would take care of adding the quote for you).

Reply:
I'd rather make the quotes explicit, relying on repr to add the quotes is less obvious.